wolfcrypt/src/asn.c

```diff
@@ -22213,11 +22214,41 @@ int wc_InitCert_ex(Cert* cert, void* heap, int devId)
     return 0;
 }
 
+WOLFSSL_ABI
 int wc_InitCert(Cert* cert)
 {
     return wc_InitCert_ex(cert, NULL, INVALID_DEVID);
 }
 
+WOLFSSL_ABI
+Cert* wc_CertNew(void* heap)
+{
+    Cert* certNew;
+
+    certNew = (Cert*)XMALLOC(sizeof(Cert), heap, DYNAMIC_TYPE_CERT);
+
+    if (certNew) {
+        if (wc_InitCert_ex(certNew, heap, INVALID_DEVID) != 0) {
+            XFREE(certNew, heap, DYNAMIC_TYPE_CERT);
+            certNew = NULL;
+        }
+    }
+
+    return certNew;
+}
+
+WOLFSSL_ABI
+void  wc_CertFree(Cert* cert)
+{
+    if (cert) {
+         void* heap = cert->heap;
+
+         ForceZero(cert, sizeof(Cert));
+         XFREE(cert, heap, DYNAMIC_TYPE_CERT);
```

Comment:
why copy `cert->heap` to `heap`, and then have to `(void)heap` in case the `XFREE()` macro ignores it?

not worth another spin through Jenkins but worth cleaning up later.

Reply:
ACKnowledged